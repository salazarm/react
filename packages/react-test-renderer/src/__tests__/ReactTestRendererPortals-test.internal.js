/**
 * Copyright (c) 2013-present, Facebook, Inc.
 *
 * This source code is licensed under the MIT license found in the
 * LICENSE file in the root directory of this source tree.
 *
 * @emails react-core
 */

'use strict';

const React = require('react');
const ReactDOM = require('react-dom');

// Isolate test renderer.
jest.resetModules();
jest.mock('shared/ReactFeatureFlags', () =>
  require('shared/forks/ReactFeatureFlags.test-renderer'),
);
const ReactTestRenderer = require('react-test-renderer');

describe('ReactTestRenderer', () => {
  it('should support ReactDOM.createPortal', () => {
    const container = document.createElement('div');
    let rendered = ReactTestRenderer.create(
      <div>
        {ReactDOM.createPortal(<span>ReactDOM portal</span>, container)}
      </div>,
    );
    expect(rendered.toJSON()).toMatchSnapshot();

    rendered.update(
      <div>
        <span>ReactTestRenderer span</span>
        {ReactDOM.createPortal(<span>ReactDOM portal</span>, container)}
      </div>,
    );
    expect(rendered.toJSON()).toMatchSnapshot();
  });
});
