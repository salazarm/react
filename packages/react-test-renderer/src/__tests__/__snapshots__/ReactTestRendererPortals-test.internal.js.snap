// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ReactTestRenderer should support ReactDOM.createPortal 1`] = `
<div>
  <span>
    ReactDOM portal
  </span>
</div>
`;

exports[`ReactTestRenderer should support ReactDOM.createPortal 2`] = `
<div>
  <span>
    ReactTestRenderer span
  </span>
  <span>
    ReactDOM portal
  </span>
</div>
`;
